ath79: support for TP-Link EAP225 v4

svenuniku opened the pull request:
These commits add support for TP-Link EAP225 v4. The main difference between v3 and v4 is the PHY `RTL8211FS`.
The second commit tries to circumvents a design decision of TP-Link:
The reset line of the PHY is not connected. Using the external Watchdog as reset line for the whole system allows the PHY to work after a reboot.

svanheule reviewed (commented):
In the device details:
* MU-MINO -> MU-MIMO
* No need to mention GPIO11 isn't connected to the phy's reset (nor are the other GPIO lines). You could just say along the lines of "the RTL8211F's reset line does not appear to be connected to the SoC".

If you copied the debricking instructions from the EAP225 v3 commit, did you check the PCB part numbers are still the same for this device?

You can also drop the `and ssid "..."`; not really adding much info with that.

You must also add a Signed-off-by line to all your commits. See https://openwrt.org/submitting-patches for guidelines, and use scripts/checkpatch.pl to verify that your patches are OK.

For example, to check one patch:
```sh
git format-patch -1 --stdout HEAD | ./scripts/checkpatch.pl
```

svenuniku committed:
ath79: support for TP-Link EAP225 v4

This model is almost identical to the EAP225 v3.
Major difference is the RTL8211FS PHY Chipset.

Device specifications:
* SoC: QCA9563 @ 775MHz
* RAM: 128MiB DDR2
* Flash: 16MiB SPI-NOR
* Wireless 2.4GHz (SoC): b/g/n, 3x3
* Wireless 5Ghz (QCA9886): a/n/ac, 2x2 MU-MIMO
* Ethernet (RTL8211FS): 1× 1GbE, 802.3at PoE

Flashing instructions:
* ssh into target device and run `cliclientd stopcs`
* Upgrade with factory image via web interface

Debricking:
* Serial port can be soldered on PCB J4 (1: TXD, 2: RXD, 3: GND, 4: VCC)
    * Bridge unpopulated resistors R225 (TXD) and R237 (RXD).
      Do NOT bridge R230.
    * Use 3.3V, 115200 baud, 8n1
* Interrupt bootloader by holding CTRL+B during boot
* tftp initramfs to flash via LuCI web interface
    setenv ipaddr 192.168.1.1 # default, change as required
    setenv serverip 192.168.1.10 # default, change as required
    tftp 0x80800000 initramfs.bin
    bootelf $fileaddr

MAC addresses:
MAC address (as on device label) is stored in device info partition at
an offset of 8 bytes. ath9k device has same address as ethernet, ath10k
uses address incremented by 1.

Signed-off-by: Sven Hauer <sven.hauer+github@uniku.de>

svenuniku commented:
@svanheule backported soft reset for the PHY. The PHY now works after a reboot without the `gpio-restart` module. See openwrt/openwrt@7bf73e270e7f1704b21860712820dcf12b593e91.

svanheule commented:
Merged with commit 7e21ce8e2b0b73f145af96958511f761544942d7. Thanks!
